posts/auditing-using-javers.md

```diff
@@ -0,0 +1,239 @@
+---
+authors:
+- Amine Ouali Alami
+- Pooja Krishnan
+tags:
+- HIPAA
+- PHI
+- Audit Trail
+- JaVers
+- Spring Data
+- Spring Boot
+- JHipster
+
+date: 2020-01-10T20:16:10.000Z
+title: "Audit your data with JaVers"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/01/Ippon_Javers.jpg
+---
+
+As an IT consultant, the first requirement that comes to mind when you are working in the Healthcare Industry is the Health Insurance Portability and Accountability Act (HIPAA). The technology plays a significant role in ensuring data remains secure and HIPAA-compliant. According to cybercrime experts and the [FBI](http://www.illuminweb.com/wp-content/uploads/ill-mo-uploads/103/2418/health-systems-cyber-intrusions.pdf), stolen Protected Health Information (PHI) is ten times more valuable than credit card information.
+
+The data contained in PHI can be used to generate identities, obtain health services, submit false insurance claims, order pharmaceuticals, and perform other illegal acts. That is why auditing PHI is critical: the ability to show who, when, and how the data is used.
+
+On the application layer, the two primary tools for data auditing are Envers and JaVers.
+- Envers is a core Hibernate module. It provides an easy auditing solution when using Hibernate or JPA. Envers is not an option if you are using a NoSQL database.
+- JaVers is a lightweight Java library for auditing changes in your data. It can be used with any kind of database and any kind of persistence framework since the audit data is easily decoupled from live data; we can choose where to store it.
+
+
+# Audit trail with JaVers
+Since our client is mainly using MongoDB, we chose to use JaVers.
+JaVers is an audit log framework that helps to track changes of entities in the application.
+
+The usage of this tool is not limited to debugging and auditing only. It can be successfully applied to perform analysis and force security policies.
+
+# Spring Boot Integration and Setup
+JaVers provides a Spring Boot starter that integrates all the required JaVers beans with the default configuration.
+
+pom.xml
+```xml
+<dependency>
+            <groupId>org.javers</groupId>
+            <artifactId>javers-spring-boot-starter-mongo</artifactId>
+            <version>5.8.8</version>
+</dependency>
+```
+
+
+# JaVers repository configuration
+JaVers [Spring Boot starter](https://javers.org/documentation/spring-boot-integration/) is configured to reuse the application's persistent data store.  This is managed by Spring Data.
+Alternatively, we can choose to store JaVers data in a separate database, as shown below:
+
+```yaml
+javers:
+  mongodb:
+    host: localhost
+    port: 27017
+    database: javers-audit
+    authentication-database: admin
+    username: javers
+    password: password
+```
+Or
+
+```yaml
+javers:
+  mongodb:
+    uri: mongodb://javers:password@localhost:27017/javers-audit&authSource=admin
+```
+
+# Auto audit annotations
```

Comment:
```suggestion
# Automatically audit annotations
```

Reply:
JaVers official documentation refers to that as 'Auto-audit annotations'.
I want to use the same expression.